src/iface/ethernet.rs

```diff
@@ -416,70 +439,73 @@ impl<'b, 'c, DeviceT> Interface<'b, 'c, DeviceT>
-            if !socket.meta_mut().egress_permitted(|ip_addr|
-                    self.inner.has_neighbor(&ip_addr, timestamp)) {
-                continue
+            if !socket
+                .meta_mut()
+                .egress_permitted(|ip_addr| self.inner.has_neighbor(&ip_addr, timestamp))
+            {
+                continue;
             }
 
             let mut neighbor_addr = None;
             let mut device_result = Ok(());
-            let &mut Self { ref mut device, ref mut inner } = self;
+            let &mut Self {
+                ref mut device,
+                ref mut inner,
+            } = self;
 
             macro_rules! respond {
-                ($response:expr) => ({
+                ($response: expr) => {{
                     let response = $response;
                     neighbor_addr = response.neighbor_addr();
                     let tx_token = device.transmit().ok_or(Error::Exhausted)?;
                     device_result = inner.dispatch(tx_token, timestamp, response);
                     device_result
-                })
+                }};
             }
 
-            let socket_result =
-                match *socket {
-                    #[cfg(feature = "socket-raw")]
-                    Socket::Raw(ref mut socket) =>
-                        socket.dispatch(&caps.checksum, |response|
-                            respond!(Packet::Raw(response))),
-                    #[cfg(all(feature = "socket-icmp", feature = "proto-ipv4"))]
-                    Socket::Icmp(ref mut socket) =>
-                        socket.dispatch(&caps, |response| {
-                            match response {
-                                #[cfg(feature = "proto-ipv4")]
-                                (IpRepr::Ipv4(ipv4_repr), icmpv4_repr) =>
-                                    respond!(Packet::Icmpv4((ipv4_repr, icmpv4_repr))),
-                                _ => Err(Error::Unaddressable)
-                            }
-                        }),
-                    #[cfg(feature = "socket-udp")]
-                    Socket::Udp(ref mut socket) =>
-                        socket.dispatch(|response|
-                            respond!(Packet::Udp(response))),
-                    #[cfg(feature = "socket-tcp")]
-                    Socket::Tcp(ref mut socket) =>
-                        socket.dispatch(timestamp, &caps, |response|
-                            respond!(Packet::Tcp(response))),
-                    Socket::__Nonexhaustive(_) => unreachable!()
-                };
+            let socket_result = match *socket {
+                #[cfg(feature = "socket-raw")]
+                Socket::Raw(ref mut socket) => {
+                    socket.dispatch(&caps.checksum, |response| respond!(Packet::Raw(response)))
+                }
+                #[cfg(all(feature = "socket-icmp", feature = "proto-ipv4"))]
+                Socket::Icmp(ref mut socket) => socket.dispatch(&caps, |response| match response {
```

Comment:
This can be tuned with [force_multiline_blocks](https://github.com/rust-lang-nursery/rustfmt/blob/master/Configurations.md#force_multiline_blocks)